### ISSUE TITLE
Bootstrapper hangs when detecting FilesInUse

### ISSUE THREAD

**Issue by wixbot** [wixtoolset/issues#4831]
I am using WiX Toolset v3.10.0.1726

I have a .msi-file that gets bundled within a bootstrapper.

MSI Defintion:

```
<Wix xmlns="http://schemas.microsoft.com/wix/2006/wi">
  <Product Id="*"
           Name="$(var.ProductName)"
           Language="!(loc.ProductLanguage)"
           Version="!(bind.fileVersion.FILE)"
           Manufacturer="$(var.Manufacturer)"
           UpgradeCode="$(var.UpgradeCode)">
    <Package InstallerVersion="200"
             Compressed="yes"
             InstallScope="perMachine"
             Platform="x86"
             InstallPrivileges="elevated"
             ReadOnly="yes" />
    <MediaTemplate EmbedCab="yes" />

    <?include Feature.wxi?>

    <MajorUpgrade AllowDowngrades="no"
                  AllowSameVersionUpgrades="yes"
                  DowngradeErrorMessage="!(loc.DowngradeErrorMessage)" />

    <UI Id="UserInterface">
      <Property Id="WIXUI_INSTALLDIR"
                Value="TARGETDIR" />
      <Property Id="WixUI_Mode"
                Value="Custom" />

      <TextStyle Id="WixUI_Font_Normal"
                 FaceName="Tahoma"
                 Size="8" />
      <TextStyle Id="WixUI_Font_Bigger"
                 FaceName="Tahoma"
                 Size="9"
                 Bold="yes" />
      <TextStyle Id="WixUI_Font_Title"
                 FaceName="Tahoma"
                 Size="9"
                 Bold="yes" />

      <Property Id="DefaultUIFont"
                Value="WixUI_Font_Normal" />

      <DialogRef Id="ProgressDlg" />
      <DialogRef Id="ErrorDlg" />
      <DialogRef Id="FilesInUse" />
      <DialogRef Id="FatalError" />
      <DialogRef Id="UserExit" />

      <Publish Dialog="ProgressDlg"
               Control="Next"
               Event="EndDialog"
               Value="Return"
               Order="2" />
    </UI>
  </Product>
</Wix>
```

If executing the .msi-file directly, `FilesInUse` dialog shows up correctly if the application is running. Everything is working fine.

Now I've integrated the .msi-file in a bootstrapper:

```
<Wix xmlns="http://schemas.microsoft.com/wix/2006/wi"
     xmlns:bal="http://schemas.microsoft.com/wix/BalExtension">
  <Bundle Name="!(bind.packageName.Setup)"
          Version="!(bind.packageVersion.Setup)"
          Manufacturer="!(bind.packageManufacturer.Setup)"
          UpgradeCode="$(var.UpgradeCode)"
          DisableModify="yes">
    <BootstrapperApplicationRef Id="WixStandardBootstrapperApplication.HyperlinkLicense">
      <bal:WixStandardBootstrapperApplication LicenseUrl=""
                                              SuppressOptionsUI="no"
                                              ShowVersion="yes"
                                              SuppressRepair="yes" />
    </BootstrapperApplicationRef>
    <Chain DisableSystemRestore="yes">
      <MsiPackage Id="Setup"
                  Compressed="yes"
                  Name="$(var.ProductName)"
                  SourceFile="setup.msi"
                  DisplayInternalUI="yes"
                  Vital="yes" />
    </Chain>
  </Bundle>
</Wix>
```

I need `DisplayInternalUI` as I am having some extra dialogs in the .msi-file which are mandatory for the setup.

If the application is not running in an upgrade scenario everything is working fine, whereas a running application while upgrading brings the bootstrapper and the .msi-file into a not responding state.

The output of the log:

> [2A34:28F0][2015-07-21T16:38:40]i001: Burn v3.10.0.1726, Windows v6.1
> (Build 7601: Service Pack 1), path: W:\work\SETUP.exe<br /> ...<br />
> [2A34:28F0][2015-07-21T16:38:40]i009: Command Line: '-burn.unelevated
> BurnPipe.{571E9E94-4908-45EE-A6AC-05F923FAFDCF}
> {F7B2FBA0-BAEB-42C7-B1D5-B8704F153049} 10904 /log log.txt'<br />
> ...<br /> [2A34:28F0][2015-07-21T16:38:41]i300: Apply begin<br />
> ...<br /> [2A98:054C][2015-07-21T16:38:42]i301: Applying execute
> package: Setup, action: Install, path: C:\ProgramData\Package
> Cache{99D06386-9447-4D7B-8307-E53C12BB09B5}v2.0.19.28124\FOOAPPLICATION,
> arguments: ' ARPSYSTEMCOMPONENT="1" MSIFASTINSTALL="7"
> INSTALLFOLDER="C:\FOOFOLDER" TRANSFORMS=":1031"'<br />

And the output of the msi-log (at least the last relevant lines):

> ...<br />
> MSI (s) (24:D4) [16:38:42:826]: Doing action: InstallValidate<br />
> MSI (s) (24:D4) [16:38:42:827]: PROPERTY CHANGE: Deleting
> MsiRestartManagerSessionKey property. Its current value is
> '053651647df31e46971d3237c9eb6a06'.<br /> MSI (s) (24:D4)
> [16:38:42:827]: Transforming table Dialog.<br /> ...<br /> MSI (s)
> (24:D4) [16:38:42:828]: Note: 1: 2205 2:  3: BindImage<br /> MSI (s)
> (24:D4) [16:38:42:828]: Note: 1: 2205 2:  3: ProgId<br /> MSI (s)
> (24:D4) [16:38:42:828]: Note: 1: 2205 2:  3: PublishComponent<br />
> MSI (s) (24:D4) [16:38:42:828]: Note: 1: 2205 2:  3: SelfReg<br /> MSI
> (s) (24:D4) [16:38:42:828]: Note: 1: 2205 2:  3: Extension<br /> MSI
> (s) (24:D4) [16:38:42:828]: Note: 1: 2205 2:  3: Font<br /> MSI (s)
> (24:D4) [16:38:42:828]: Note: 1: 2205 2:  3: Class<br /> MSI (s)
> (24:D4) [16:38:42:828]: Note: 1: 2205 2:  3: TypeLib<br /> MSI (s)
> (24:D4) [16:38:42:829]: Note: 1: 2205 2:  3: _RemoveFilePath<br /> MSI
> (s) (24:D4) [16:38:42:993]: Note: 1: 1402 2: HKEY_CLASSES_ROOT.NET 3:
> 2<br /> MSI (s) (24:D4) [16:38:43:127]: Note: 1: 1402 2:
> HKEY_CLASSES_ROOT.NET 3: 2<br /> MSI (s) (24:D4) [16:38:43:221]:
> PROPERTY CHANGE: Modifying CostingComplete property. Its current value
> is '0'. Its new value: '1'.<br /> MSI (s) (24:D4) [16:38:43:222]:
> Note: 1: 2205 2:  3: BindImage<br /> MSI (s) (24:D4) [16:38:43:222]:
> Note: 1: 2205 2:  3: ProgId<br /> MSI (s) (24:D4) [16:38:43:222]:
> Note: 1: 2205 2:  3: PublishComponent<br /> MSI (s) (24:D4)
> [16:38:43:222]: Note: 1: 2205 2:  3: SelfReg<br /> MSI (s) (24:D4)
> [16:38:43:222]: Note: 1: 2205 2:  3: Extension<br /> MSI (s) (24:D4)
> [16:38:43:222]: Note: 1: 2205 2:  3: Font<br /> MSI (s) (24:D4)
> [16:38:43:222]: Note: 1: 2205 2:  3: Class<br /> MSI (s) (24:D4)
> [16:38:43:222]: Note: 1: 2205 2:  3: TypeLib<br /> MSI (s) (24:D4)
> [16:38:43:222]: Note: 1: 2727 2:<br /> MSI (s) (24:D4) [16:38:43:314]:
> Transforming table Error.<br /> <br /> MSI (s) (24:D4) [16:38:43:314]:
> Transforming table Error.<br /> <br /> MSI (c) (98:4C) [16:38:43:326]:
> RESTART MANAGER: Session opened.

So, why does the msi in the bootstrapped scenario hang (and causes the bootstrapper to hang either), and how do I fix this?

> Originally opened by `andreas`

**Comment by aleksandr-grechko** [comment]
Hi,
We need to check if current package shows internal UI before display the FilesInUse page.
I would like to fix this defect.

**Comment by aleksandr-grechko** [comment]
My fix ([#408](https://github.com/wixtoolset/wix3/pull/408)) was merged to Wix3 by @rseanhall. Currently we don't have FileInUse functional for WixStdBA  in Wix4. So probably this issue can be closed.
